CONTRIBUTING.md

```diff
@@ -39,4 +39,32 @@ If you need to make changes to the pull request, simply repeat steps 2-4.  Addin
 
 The majority of code in MonoGame is cross-platform and must build and behave correctly on all supported platforms.  All pull requests to MonoGame are built using an automated system that compiles the pull request on all supported platforms and will report any build errors.
 
-Once a pull request has been accepted, your feature branch can be deleted if desired.
\ No newline at end of file
+Once a pull request has been accepted, your feature branch can be deleted if desired.
+
+Generating Documentation
+========================
+Monogame generates its documentation using the [SharpDoc library] (https://github.com/xoofx/SharpDoc)
+
+You can view the documentation for your changes both locally, and (after you have submitted a PR) on the build server.
+
+### Local documentation generation
+Local documentation generation is not automatic. You have to run the SharpDoc tool to generate it for you.
+The commandline for doing this is: `ThirdParty\Dependencies\SharpDoc\SharpDoc.exe -config <config file> -output <output directory>`
+e.g. `ThirdParty\Dependencies\SharpDoc\SharpDoc.exe -config Documentation\config.xml -output GeneratedDocs`
+Run this command from the root folder of your local MonoGame repository clone. It will generate all the documentation under `Documentation\<output directory>` folder. You can go ahead and start browsing from the `index.html` file.
+
+**Gotchas:**
```

Comment:
You mean "Gotchas" right?
That wasn't actually a header, just bold text. Came out looking like a h3. Don't know why. Will fix that too